plugins/plugin-chart-echarts/src/Pie/EchartsPie.tsx

```diff
@@ -20,49 +20,71 @@ import React, { useCallback } from 'react';
 import { PieChartTransformedProps } from './types';
 import Echart from '../components/Echart';
 import { EventHandlers } from '../types';
+import { DataMask, DrillDown } from "@superset-ui/core";
 
 export default function EchartsPie({
+  formData,
   height,
   width,
   echartOptions,
   setDataMask,
   labelMap,
   groupby,
   selectedValues,
-  formData,
+  ownState,
 }: PieChartTransformedProps) {
   const handleChange = useCallback(
     (values: string[]) => {
-      if (!formData.emitFilter) {
+      if (!formData.emitFilter && !formData.drillDown) {
         return;
       }
 
       const groupbyValues = values.map(value => labelMap[value]);
 
-      setDataMask({
-        extraFormData: {
-          filters:
-            values.length === 0
-              ? []
-              : groupby.map((col, idx) => {
-                  const val = groupbyValues.map(v => v[idx]);
-                  if (val === null || val === undefined)
+      let dataMask: DataMask = {};
+      if (formData.emitFilter) {
+        dataMask = {
+          extraFormData: {
+            filters:
+              values.length === 0
+                ? []
+                : groupby.map((col, idx) => {
+                    const val = groupbyValues.map(v => v[idx]);
+                    if (val === null || val === undefined)
```

Comment:
As I remember, in JS better to always use `{}` in `if` statement even if we have a one action. So we can use `if (val === null || val === undefined) { ...do something }`
Or `return (val === null || val === undefined) ? { col, op: 'IS NULL' } : { col, op: 'IN' ... } `

Also about `=== null` checks, we can use !val instead both checks, so `return !val ? { col, op: 'IS NULL' } : { col, op: 'IN' ... } `

Reply:
Also about this `const val = groupbyValues.map(v => v[idx]);`: as I remember, .map() will return an array so maybe will be better to use `!val.length` if we need to check, is val contains something

Reply:
yeah you're probably right about all these points.  on this i was blindly copying from the referenced PR.  i'll take a look and make the changes you suggest unless something goes wrong.

Reply:
interestingly, everything you're pointing out is in base superset.

Reply:
so i stared at this a lot... there's no way for val to be anything but an array, so the existing `if (val === null || val === undefined)` can't ever actually do anything.  i've removed that, which should address all of your style-related comments.